include/boost/math/tools/polynomial.hpp

```diff
@@ -290,6 +290,12 @@ class polynomial :
           m_data.push_back(point);
```

Comment:
Should we remove the `explicit` keyword from this constructor? I believe that it would provide us with opportunities to simplify the code and shave off some lines.